eshell-info-banner.el

```diff
@@ -292,20 +292,46 @@ the warning face with a battery level of 25% or less."
                                      (number-to-string percentage)
                                      :inherit (eshell-info-banner--get-color-percentage (- 100.0 percentage)))))))))
 
+                                        ; Operating system identification ;;;;;;;;;;;;;;;;;;
+(defun eshell-info-banner--get-os-information-from-release-file (&optional release-file)
+  "Read the operating system from the given RELEASE-FILE.
+
+If RELEASE-FILE is nil, use '/etc/os-release'."
+  (setq release-file (or release-file "/etc/os-release"))
+  (replace-regexp-in-string
+   ".*\"\\(.+\\)\""
+   "\\1"
+   (car (-filter (lambda (line)
+                   (s-contains? "PRETTY_NAME" line))
+                 (s-lines (with-temp-buffer
+                            (insert-file-contents release-file)
+                            (buffer-string)))))))
+
+(defun eshell-info-banner--get-os-information-from-hostnamectl ()
+  "Read the operating system via hostnamectl."
+  (with-temp-buffer
+    (call-process "hostnamectl" nil t nil)
+    (re-search-backward "Operating System: \\(.*\\)")
+    (match-string 1)))
+
+(defun eshell-info-banner--get-os-information-from-lsb-release ()
+  "Read the operating system information from lsb_release"
+  (shell-command-to-string "lsb_release -d -s"))
+
+(defun eshell-info-banner--get-os-information ()
+  "Get operating system identifying information."
+  (or (ignore-errors (eshell-info-banner--get-os-information-from-hostnamectl))
+      (ignore-errors (eshell-info-banner--get-os-information-from-lsb-release))
+      (ignore-errors (eshell-info-banner--get-os-information-from-release))
+      "Unkown"))
```

Comment:
I think something that could be done here instead of `ignore-errors` is to make the three functions return a string only if the related executable or file is found, otherwise return `nil`. Something like
```emacs-lisp
(when (executable-find "hostnamectl")
  ;; return stuff
  )
```
That way, you won’t need to use `ignore-errors`.

Reply:
Yeah, I mentioned a `cond` variant in the PR message and can use that one instead, if you want to.